activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -523,6 +521,10 @@ def add_index(table_name, column_name, options = {}) #:nodoc:
       end
 
       def foreign_keys(table_name)
+        raise ArgumentError unless table_name.present?
```

Comment:
@jeremy said he prefer to raise an ArgumentError for an exceptional case like this in https://github.com/rails/rails/pull/21614#discussion_r39651518